src/features/AddAccount/components/HDWallets.tsx

```diff
@@ -218,28 +210,24 @@ export function HDWalletsClass({
     return;
   }, [requestingBalanceCheck]);
 
-  const getAddresses = (props: OwnProps) => {
+  const getAddresses = async () => {
     setRequestingWallets(true);
-    // tslint:disable-next-line: no-shadowed-variable
-    const { dPath, publicKey, chainCode, seed } = props;
-    if (dPath && ((publicKey && chainCode) || seed)) {
-      if (isValidPath(dPath.value)) {
-        setWallets(
-          getHDWallets({
-            seed,
-            dPath: dPath.value,
-            publicKey,
-            chainCode,
-            limit: WALLETS_PER_PAGE,
-            offset: WALLETS_PER_PAGE * page
-          })
-        );
+    if (dPath && isValidPath(dPath.path)) {
+      try {
+        const result = await wallet!.getAddresses({
+          path: dPath,
+          limit: WALLETS_PER_PAGE,
+          offset: WALLETS_PER_PAGE * page
+        });
+        setWallets(result);
         setRequestingWallets(false);
         setRequestingBalanceCheck(true);
-      } else {
-        setRequestingWallets(false);
-        setRequestingBalanceCheck(false);
+      } catch (err) {
+        console.error(err);
```

Comment:
We don't really have an error state for this. Before it would just be an uncaught error. Now we at least catch it 🤔 

Haven't spent much time on it, since it is the legacy component.